### DISCUSSION TITLE
implicit conversions

### DISCUSSION THREAD

**Discussion by huangweiwu** [wjakob/nanobind#257] [Q&A] [answered]
Int cannot implicit convert to bool?

void test_bool(bool a)
{
}
NB_MODULE(my_ext, m) {
    m.def("test_bool", test_bool);
}
when I call test_bool(1) in python, a TypeError happens.
TypeError: test_bool(): incompatible function arguments. The following argument types are supported:
    1. test_bool(arg: bool, /) -> None

**Comment by wjakob** [comment, marked as answer]
nanobind's bool conversion is stricter than the "truthiness" rules in Python. You could modify the type caster to make the behavior more relaxed. The stricter builtin behavior is intentional and I do not wish to change it.

**Reply by huangweiwu** [reply]
Got it. Thanks a lot.
